### PR TITLE
Removing the route timeout validation

### PR DESCRIPTION
**Pull request by kobelb**
Resolves https://github.com/hapijs/hapi/issues/4122

### CODE REVIEW

File: lib/route.js [hunk 1]
```diff
@@ -363,9 +358,7 @@ exports = module.exports = internals.Route = class {
             return;
         }
 
-        if (this.method[0] !== '_') {
```
Comment (kobelb) [inline lib/route.js:366]: This conditional is no longer needed, and having it here reduced the code-coverage.

File: test/route.js [hunk 2]
```diff
@@ -450,20 +450,20 @@ describe('Route', () => {
         expect(res.payload).to.contain('hapi');
     });
 
-    it('throws when server timeout is more then socket timeout', () => {
+    it('allows server timeout more then socket timeout', () => {
```
Comment (cjihrig) [inline test/route.js:453]: We should probably have similar tests for the payload timeout.

File: lib/route.js (hunk 1 above)
Comment (cjihrig) [reply lib/route.js:366]: Is this actually not needed anymore, or just not exercised by the tests anymore?

Comment (kobelb) [reply lib/route.js:366]: I haven't been able to find a code-path that relies on this branch, but I have to admit I'm not fluent in Hapi's codebase.

Comment (devinivy) [reply lib/route.js:366]: The method used for the internal not found, bad request, and cors preflight routes is `_special`.  I think we'll want to keep this around and find a way to exercise it with one of those routes.

File: test/route.js (hunk 2 above)
Comment (kobelb) [reply test/route.js:485]: Done!

File: lib/route.js (hunk 1 above)
Comment (kobelb) [reply lib/route.js]: You all were right. I found a way to exercise this code-path with https://github.com/hapijs/hapi/pull/4123/commits/3dc68cc90f713e35e72dcb23a84665c3a2778ca7

**Comment by kobelb** [comment]
Assuming others approve this PR, which versions will include this fix? Is it possible to have it included in v17?

**Comment by cjihrig** [comment]
If this lands, it will only be for v20. If you need support for an older version, you'll have to check on a commercial license - https://github.com/hapijs/hapi/issues/4114.

**Comment by cjihrig** [comment]
Thanks for the contribution!

**Pull request merged**